test/integration/aggregate.test.ts

```diff
@@ -0,0 +1,313 @@
+    const result = await xata.db.teams.aggregate({
+      total: { count: '*' },
+      unique: { uniqueCount: { column: 'name' } }
+    });
+
+    expect(result.aggs.total).toBe(5);
+    expect(result.aggs.unique).toBe(3);
+  });
+
+  test('count and unique count with global filter', async () => {
+    const result = await xata.db.teams.aggregate(
+      {
+        total: { count: '*' },
+        unique: { uniqueCount: { column: 'name' } }
+      },
+      { 'settings.plan': 'free' }
+    );
+
+    expect(result.aggs.total).toBe(3);
+    expect(result.aggs.unique).toBe(2);
+  });
+
+  test('counts with filters', async () => {
+    const result = await xata.db.teams.aggregate({
+      freeCount: { count: { filter: { 'settings.plan': 'free' } } },
+      paidCount: { count: { filter: { 'settings.plan': 'paid' } } }
+    });
+
+    expect(result.aggs.freeCount).toBe(3);
+    expect(result.aggs.paidCount).toBe(2);
+  });
+
+  test('numeric stats', async () => {
+    const result = await xata.db.teams.aggregate({
+      sumIndex: { sum: { column: 'index' } },
+      sumRating: { sum: { column: 'rating' } },
+      maxIndex: { max: { column: 'index' } },
+      maxRating: { max: { column: 'rating' } },
+      minIndex: { min: { column: 'index' } },
+      minRating: { min: { column: 'rating' } },
+      averageIndex: { average: { column: 'index' } },
+      averageRating: { average: { column: 'rating' } }
+    });
+
+    expect(result.aggs.sumIndex).toBe(31);
+    expect(result.aggs.sumRating).toBe(15.6);
+    expect(result.aggs.maxIndex).toBe(12);
+    expect(result.aggs.maxRating).toBe(5.3);
+    expect(result.aggs.minIndex).toBe(2);
+    expect(result.aggs.minRating).toBe(1);
+    expect(result.aggs.averageIndex).toBe(6.2);
+    expect(result.aggs.averageRating).toBe(3.12);
+  });
+
+  test('simple date histogram', async () => {
+    const result = await xata.db.teams.aggregate({
+      byDate: {
+        dateHistogram: {
+          column: 'founded_date',
+          calendarInterval: 'week'
+        }
+      }
+    });
+
+    expect(result.aggs.byDate.values).toHaveLength(3);
+    expect(result.aggs.byDate.values?.[0].$key).toBe('2021-12-27T00:00:00.000Z');
+    expect(result.aggs.byDate.values?.[0].$count).toBe(2);
+    expect(result.aggs.byDate.values?.[1].$key).toBe('2022-01-03T00:00:00.000Z');
+    expect(result.aggs.byDate.values?.[1].$count).toBe(2);
+    expect(result.aggs.byDate.values?.[2].$key).toBe('2022-01-10T00:00:00.000Z');
+    expect(result.aggs.byDate.values?.[2].$count).toBe(1);
+  });
+
+  test('date histogram with sub-aggs', async () => {
+    const result = await xata.db.teams.aggregate({
+      byDate: {
+        dateHistogram: {
+          column: 'founded_date',
+          calendarInterval: 'week',
+          aggs: {
+            total: { count: '*' },
+            unique: { uniqueCount: { column: 'name' } }
+          }
+        }
+      }
+    });
+
+    expect(result.aggs.byDate.values).toHaveLength(3);
```

Comment:
Nitpick. In these tests we could extract this to a variable to make the other lines a bit more DRY.

```ts
    const values = result.aggs.byDate.values;
```

Reply:
I'll clean the tests a bit in a follow up PR.

Being honest the test have mainly been written by Copilot translating them from the backend ones in go 😍🙃

I just gave it as input the type definition of the method and the input/output values of all the test cases, and it correctly generated function calls with the correct expects.

I love the future